nomad/state/testing.go

```diff
@@ -16,8 +16,9 @@ import (
 
 func TestStateStore(t testing.TB) *StateStore {
 	config := &StateStoreConfig{
-		Logger: testlog.HCLogger(t),
-		Region: "global",
+		Logger:             testlog.HCLogger(t),
+		Region:             "global",
+		JobTrackedVersions: structs.JobTrackedVersions,
```

Comment:
As noted above, there's no test case exercising the change in behavior (in particular around the 0 case).

Reply:
We didn't add any new tests for this behavior, but now that the 0 value is not allowed I think we're ok. I did manually test out the behavior of `auto_revert = true` with `job_tracked_versions = 1` and it seems correct and reasonable.